From c4a01182b37256b21a73f60a1cf32120a0c7da69 Mon Sep 17 00:00:00 2001
From: Lin Jinhan <troy.lin@rock-chips.com>
Date: Fri, 14 Aug 2020 09:40:36 +0800
Subject: [PATCH] crypto: rockchip: drop needless dma_sync_single_for_device

hw_info->desc is allocated by dma_alloc_coherent, there is needless
to call dma_sync_single_for_device.

Change-Id: Ibd6c834c6f772727a829381dd4b0c95ceb39e5cd
Signed-off-by: Lin Jinhan <troy.lin@rock-chips.com>
---
 drivers/crypto/rockchip/rk_crypto_v2_ablkcipher.c | 2 --
 drivers/crypto/rockchip/rk_crypto_v2_ahash.c      | 2 --
 2 files changed, 4 deletions(-)

diff --git a/drivers/crypto/rockchip/rk_crypto_v2_ablkcipher.c b/drivers/crypto/rockchip/rk_crypto_v2_ablkcipher.c
index f2d61ba09d55..9a09cf9ac8ec 100644
--- a/drivers/crypto/rockchip/rk_crypto_v2_ablkcipher.c
+++ b/drivers/crypto/rockchip/rk_crypto_v2_ablkcipher.c
@@ -389,8 +389,6 @@ static void crypto_dma_start(struct rk_crypto_info *dev)
 	hw_info->desc->dma_ctrl = 0x00000201;
 	hw_info->desc->user_define = 0x7;
 
-	dma_sync_single_for_device(dev->dev, hw_info->desc_dma,
-				   sizeof(*hw_info->desc), DMA_TO_DEVICE);
 	CRYPTO_WRITE(dev, CRYPTO_DMA_LLI_ADDR, hw_info->desc_dma);
 	CRYPTO_WRITE(dev, CRYPTO_DMA_CTL, 0x00010001);/* start */
 }
diff --git a/drivers/crypto/rockchip/rk_crypto_v2_ahash.c b/drivers/crypto/rockchip/rk_crypto_v2_ahash.c
index a4459b8401a3..9ac3d63b5f28 100644
--- a/drivers/crypto/rockchip/rk_crypto_v2_ahash.c
+++ b/drivers/crypto/rockchip/rk_crypto_v2_ahash.c
@@ -433,8 +433,6 @@ static void rk_ahash_dma_start(struct rk_crypto_info *dev)
 	hw_info->desc->dma_ctrl = 0x00000401;
 	hw_info->desc->user_define = 0x7;
 
-	dma_sync_single_for_device(dev->dev, hw_info->desc_dma,
-				   sizeof(*hw_info->desc), DMA_TO_DEVICE);
 	CRYPTO_WRITE(dev, CRYPTO_DMA_LLI_ADDR, hw_info->desc_dma);
 	CRYPTO_WRITE(dev, CRYPTO_HASH_CTL,
 		     (CRYPTO_HASH_ENABLE <<
-- 
2.35.3

